src/rp2_common/pico_multicore/include/pico/multicore.h

```diff
@@ -133,38 +190,110 @@ static inline void multicore_fifo_drain(void) {
+ *
+ * \note When "locked out" the victim core is executing code from RAM and has interrupts disabled. This allows
+ * the lockout functions to be used by code that wants to write to flash (at which point no code must be executing
+ * from flash)
+ *
+ * The core which wishes to lockout the other core calls \ref multicore_lockout_start_blocking or
+ * \ref multicore_lockout_start_timeout_us to interrupt the other "victim" core and wait for it to be in a
+ * "locked out" state. Once the lockout is no longer needed it calls \ref multicore_lockout_end_blocking or
+ * \ref multicore_lockout_end_timeout_us to release the lockout and wait for confirmation.
+ *
+ * \note Because multicore lockout uses the intercore FIFOs, the FIFOs <b>cannot</b> be used for any other purpose
+ */
+
+/*! \brief Initialize the current core such that it can be a "victim" of lockout (i.e. force to pause in a known state by the other core)
+ *  \ingroup multicore_lockout
+ *
+ * This code hooks the multicore FIFO IRQ, and the FIFO may not be used for any other purpose after this.
+ */
 void multicore_lockout_victim_init(void);
 
-// start locking out the other core (it will be
-bool multicore_lockout_start_timeout_us(uint64_t timeout_us);
+/*! \brief Request the other core to paused in a known state and wait for it to do so
+ *  \ingroup multicore_lockout
+ *
+ * The other (victim) core must have previously executed \ref multicore_lockout_victim_init()
+ */
 void multicore_lockout_start_blocking(void);
 
-bool multicore_lockout_end_timeout_us(uint64_t timeout_us);
+/*! \brief Request the other core to paused in a known state and wait up to a time limit for it to do so
+ *  \ingroup multicore_lockout
+ *
+ * The other core must have previously executed \ref multicore_lockout_victim_init()
+ *
+ * \param timeout_us the timeout in microseconds
+ * \return true if the other core entered the locked out state within the timeout, false otherwise
+ */
+bool multicore_lockout_start_timeout_us(uint64_t timeout_us);
+
+/*! \brief Release the other core from a locked out state amd wait for it to acknowledge
+ *  \ingroup multicore_lockout
+ *
+ * \note The other core must previously have been "locked out" by calling a `multicore_lockout_start_` function
+ * from this core
+ */
 void multicore_lockout_end_blocking(void);
 
+/*! \brief Release the other core from a locked out state amd wait up to a time limit for it to acknowledge
+ *  \ingroup multicore_lockout
+ *
+ * The other core must previously have been "locked out" by calling a `multicore_lockout_start_` function
+ * from this core
+ *
+ * \note be very careful using small timeout values, as a timout here will leave these "lockout" functionality
+ * in a bad state. It is probably preferable to use \ref multicore_lockout_end_blocking anyway as if you have
+ * already waited for the victim core to enter the lockout state, then the victim core will be ready to exit
+ * the lockout state very quickly.
```

Comment:
Given that you can only call `multicore_lockout_end_` after `multicore_lockout_start_`, what's the use-case for ever needing/wanting to use `multicore_lockout_end_timeout_us` ?  "if you have already waited for the victim core to enter the lockout state" - there doesn't seem to be any way to **not** wait for the victim core to enter the lockout state? :thinking:

Reply:
yeah `multicore_lockout_end_timeout_us` is largely pointless, but since i can't remove it i thought id improve the docs.

most of the complication arises from the possibility of both cores trying to lock each other out the same time or so (which probably people are not actually doing)